### PR TITLE
add pinList auto pagination

### PR DESCRIPTION
**Pull request by rommel-pinata**
Compability for await ...
THIS IS A MAJOR RELEASE because it could require to use node 10

### CODE REVIEW

**Comment by rommel-pinata** [comment]
Screen Shot 2022-10-12 at 12 23 43

File: src/commands/data/getFilesByCount/getFilesByCount.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            let pageOffset = 0;
+            let cache = [];
+            let count = 0;
+            
+            return {
+                next() {
+                    return new Promise((resolve, reject) => {
+                        if (i === 0 || (i % pageLimit === 0 && i < count)) {
+                            resolve(
+                                pinList(pinataApiKey, pinataSecretApiKey, {
```
Comment (polluterofminds) [inline src/commands/data/getFilesByCount/getFilesByCount.js:28]: @rommel-pinata we need to change the way the `pinList` call works before we can do this. The pinList method needs to include a query string of `includesCount=false`

**Review by polluterofminds — CHANGES_REQUESTED** [review]
Added an inline comment. Need to make the suggested change and test it before this PR can be completed.

File: src/commands/data/getFilesByCount/getFilesByCount.js (hunk 1 above)
Comment (rommel-pinata) [reply src/commands/data/getFilesByCount/getFilesByCount.js:28]: @polluterofminds  I added the includeCount param to the pinList!

**Review by polluterofminds — CHANGES_REQUESTED** [review]
@rommel-pinata Can you add tests for the new function and update the tests for the pinList function?

**Review by polluterofminds — APPROVED** [review]

**Pull request merged**